### PR TITLE
Upgrade nghttp2

### PR DESCRIPTION
**Pull request by L3n41c**

### CODE REVIEW

File: config/software/sysstat.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name "sysstat"
-default_version "11.1.3"
+default_version "12.4.0"
```
Comment (L3n41c) [inline config/software/sysstat.rb:2]: This version of `sysstat` cannot be compiled by the version of `gcc` we use in our CI:
https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/41290288
```
gcc -o sar -L/opt/datadog-agent/embedded/lib -I/opt/datadog-agent/embedded/include -Wall -Wstrict-prototypes -pipe -O2 sar.o act_sar.o format_sar.o sa_common.o pr_stats.o librdstats_light.a libsyscom.a -s
Error:
    sa_common.c: In function ‘handle_invalid_sa_file’:
sa_common.c:343:5: warning: implicit declaration of function ‘__builtin_bswap16’ [-Wimplicit-function-declaration]
sa_common.c: In function ‘handle_invalid_sa_file’:
sa_common.c:343:5: warning: implicit declaration of function ‘__builtin_bswap16’ [-Wimplicit-function-declaration]
[…]
ar: creating libsyscom.a
sa_common_sadc.o: In function `handle_invalid_sa_file':
sa_common.c:(.text+0x466): undefined reference to `__builtin_bswap16'
collect2: error: ld returned 1 exit status
make: *** [sadc] Error 1
make: *** Waiting for unfinished jobs....
sa_common.o: In function `handle_invalid_sa_file':
sa_common.c:(.text+0x466): undefined reference to `__builtin_bswap16'
sa_common.o: In function `sa_open_read_magic':
sa_common.c:(.text+0x2078): undefined reference to `__builtin_bswap16'
collect2: error: ld returned 1 exit status
make: *** [sar] Error 1
```

File: config/software/supervisor.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name "supervisor"
-default_version "3.3.3"
+default_version "4.2.1"
```
Comment (L3n41c) [inline config/software/supervisor.rb:2]: This seems a big version bump but:
In the agent container, we are already packaging a much more recent version of `supervisor`:
```
$ docker run -ti --rm datadog/agent:7.22.1 supervisord --version
4.1.0
```

File: config/software/sysstat.rb (hunk 1 above)
Comment (L3n41c) [reply config/software/sysstat.rb]: DataDog/datadog-agent#6514

File: config/software/supervisor.rb (hunk 2 above)
Comment (L3n41c) [reply config/software/supervisor.rb]: The version of `supervisor` that is shipped with the agent is in fact coming from:
https://github.com/DataDog/integrations-core/blob/master/supervisord/requirements.in#L2
```
supervisor==4.1.0
```

So, the version in this file doesn’t matter.

**Comment by L3n41c** [comment]
Here is an agent CI build with this version of nghttp2: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/41443548/raw

File: config/software/nghttp2.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 name "nghttp2"
-default_version "1.12.0"
+default_version "1.41.0"
 
 dependency "openssl"
 
-source url: "https://github.com/nghttp2/nghttp2/releases/download/v1.12.0/nghttp2-#{version}.tar.gz",
```
Comment (hush-hush) [inline config/software/nghttp2.rb:6]: Could we keep the previous md5sum ? It make reverting version much easier.

See [this](https://github.com/DataDog/omnibus-software/blob/master/config/software/libtool.rb#L26) or [this](https://github.com/DataDog/omnibus-software/blob/master/config/software/libpng.rb#L23) for examples.

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: config/software/nghttp2.rb (hunk 3 above)
Comment (L3n41c) [reply config/software/nghttp2.rb:6]: Thanks @hush-hush for the suggestion !
I’ve implemented it.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**